cmd/tailscale/cli/debug.go

```diff
@@ -789,3 +801,31 @@ func runCapture(ctx context.Context, args []string) error {
 	_, err = io.Copy(f, stream)
 	return err
 }
+
+var debugPortmapArgs struct {
+	duration time.Duration
+	gwSelf   string
+	ty       string
+}
+
+func debugPortmap(ctx context.Context, args []string) error {
+	vals := make(url.Values)
+	vals.Set("duration", debugPortmapArgs.duration.String())
+	vals.Set("type", debugPortmapArgs.ty)
+	if debugPortmapArgs.gwSelf != "" {
+		vals.Set("gateway_and_self", debugPortmapArgs.gwSelf)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", "http://local-tailscaled.sock/localapi/v0/debug-portmap?"+vals.Encode(), nil)
```

Comment:
Can this be a `LocalClient` method (so that callers don't have to be aware of query argument mapping)? That's how all other CLI functionality that uses LocalAPI endpoints is implemented.

Reply:
Pushed; I had the method return an `io.ReadCloser` so that we can still stream logs from the server.